www/js/lib/util.js

```diff
@@ -86,12 +86,12 @@ define(['q'], function(q) {
      */
     function removeDuplicateTitlesInDirEntryArray(array) {
         array.sort(function(dirEntryA, dirEntryB) {
-            if (dirEntryA.title < dirEntryB.title) return -1;
-            if (dirEntryA.title > dirEntryB.title) return 1;
+            if (dirEntryA.getTitleOrUrl() < dirEntryB.getTitleOrUrl()) return -1;
+            if (dirEntryA.getTitleOrUrl() > dirEntryB.getTitleOrUrl()) return 1;
             return 0;
         });
         for(var i = 1; i < array.length; ){
-            if(array[i-1].title === array[i].title){
+            if(array[i-1].getTitleOrUrl() === array[i].getTitleOrUrl()){
```

Comment:
@mossroy Is this line the problem? I assumed this was an array of dirEntries, but maybe it isn't.

Reply:
Sorry, my bad : it's supposed to run on DirEntries, but the unit test does not (and it's unused in the actual code). You can leave it with .title for this PR.
Maybe we should remove this function and its unit test